src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -251,20 +252,189 @@ export class MagicWandService {
   }
 
   /**@license MIT License <http://www.opensource.org/licenses/mit-license.php>
-  * Pretty good color distance from
-  * http://www.compuphase.com/cmetric.htm
-  */
+   * Pretty good color distance from
+   * http://www.compuphase.com/cmetric.htm
+   */
   colorDistance(a: Color, b: Color): number {
     const dr = a.red - b.red;
     const dg = a.green - b.green;
     const db = a.blue - b.blue;
     const redMean = (a.red + b.red)/2;
     return (2+redMean/256)*dr*dr + 4*dg*dg + (2 + (255 - redMean)/256)*db*db;
   }
+
+
+  /* Code for 'preview' algorithm */
+
+  comparator(x: PixelNode, y: PixelNode): boolean {
+    return x.distance < y.distance;
+  }
+
+  /**Runs shortest paths algo to find shortest distance from input pixel
+   * to every other pixel.
+   * Distance is judged by the pixel's colorDistance in relation to the input 
+   * pixel.
+   * @returns Array<number> distances, with indices representing the pixel coord
+   * and values representing the shortest distance of that pixel coord.
+   */
+  getShortestPaths(imgData: ImageData, xCoord: number, yCoord: number,
+      toleranceLimit: number): Array<number> {
```

Comment:
@snazzorama Is it okay if I add a TODO here for that, and then work on that TODO in a separate branch when I get a chance? If I extend this implementation to scribbles, I would also have to rewire how the scribbles is called inside the mask-directive and editor-component, and since we have to merge all our code and branches into something that functions properly by tomorrow, then I am not 100% confident that I could get that changed, tested and decently-bug-free on time.

Reply:
Yep, for sure, a TODO is fine. If we don't get to scribbles at all for this, that's OK too; I think there are higher priorities.